test/unit/pages/patientdata.test.js

```diff
@@ -132,87 +138,81 @@ describe('PatientData', function () {
-    var elem = TestUtils.renderIntoDocument(<PatientData {...props}/>);
-    expect(elem).to.be.ok;
-    
-    var x = TestUtils.findRenderedDOMComponentWithClass(elem, 'patient-data-message');
-    expect(x).to.be.ok;
-  });
+      // Try out using the spread props syntax in JSX
+      var elem = TestUtils.renderIntoDocument(<PatientData {...props}/>);
+      expect(elem).to.be.ok;
+      
+      var x = TestUtils.findRenderedDOMComponentWithClass(elem, 'patient-data-message');
+      expect(x).to.be.ok;
+    });
 
-  it ('should render when data is present for current patient', function() {
-    var props = {
-      timePrefs: {
-        timezoneAware: false,
-        timezoneName: null
-      },
-      patient: {
-        userid: 40
-      },
-      patientData: {
-        40: { data: [ 1, 2] }
-      },
-      fetchingPatient: false,
-      fetchingPatientData: false,
-      queryParams: {},
-      trackMetric: sinon.stub()
-    };
-
-    PatientData.__set__('Modal', React.createClass({
-      render: function() {
-        console.log('Mock', this.props);
-        return <div className='fake-modal-view'/>;
-      }
-    }));
-
-    // Try out using the spread props syntax in JSX
-    var elem = TestUtils.renderIntoDocument(<PatientData {...props}/>);
-    expect(elem).to.be.ok;
-    
-    // var x = TestUtils.findRenderedDOMComponentWithClass(elem, 'fake-modal-view');
-    // expect(x).to.be.ok;
+    it ('should render the no data message when no data is present for current patient', function() {
+      var props = {
+        timePrefs: {
+          timezoneAware: false,
+          timezoneName: null
+        },
+        patient: {
+          userid: 40
+        },
+        patientData: {
+          41: { data: [] }
+        },
+        fetchingPatient: false,
+        fetchingPatientData: false,
+        queryParams: {},
+        trackMetric: sinon.stub()
+      };
+
+      // Try out using the spread props syntax in JSX
+      var elem = TestUtils.renderIntoDocument(<PatientData {...props}/>);
+      expect(elem).to.be.ok;
+      
+      var x = TestUtils.findRenderedDOMComponentWithClass(elem, 'patient-data-message');
+      expect(x).to.be.ok;
+    });
+
+    it ('should render when data is present for current patient', function() {
+      var props = {
+        timePrefs: {
+          timezoneAware: false,
+          timezoneName: null
+        },
+        patient: {
+          userid: 40
+        },
+        patientData: {
+          40: { data: [ 1, 2] }
+        },
+        fetchingPatient: false,
+        fetchingPatientData: false,
+        queryParams: {},
+        trackMetric: sinon.stub()
+      };
+
+      var pdElem = React.createElement(PatientData, props);
+      var elem = TestUtils.renderIntoDocument(pdElem);
+      expect(elem).to.be.ok;
+      
+      // var x = TestUtils.scryRenderedDOMComponentsWithClass(elem, 'fake-modal-view');
+      // console.log(x.length);
+      //expect(x).to.be.ok;
```

Comment:
I'll remove this for now as we are keen to get everything merged in. It is something I want figure out though as this will mean testing approach can be drastically improved by isolating tests just to the class you want to test. I'll work on it on a seperate branch